lib/ansible/modules/remote_management/oneview/oneview_datacenter.py

```diff
@@ -0,0 +1,170 @@
+            - resourceName: '{{ datacenter_content_rack_name }}' # option 1
+              resourceUri: ''                                    # option 2
+              x: 1000
+              y: 1000
+        depth: 6000
+        name: "MyDatacenter"
+        width: 5000
+  delegate_to: localhost
+
+- name: Update the Datacenter with specified properties (no racks)
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: present
+    data:
+        contents: ~
+        coolingCapacity: '5'
+        coolingMultiplier: '1.5'
+        costPerKilowattHour: '0.10'
+        currency: USD
+        defaultPowerLineVoltage: '220'
+        deratingPercentage: '20.0'
+        depth: 5000
+        deratingType: NaJp
+        name: "MyDatacenter"
+        width: 4000
+  delegate_to: localhost
+
+- name: Rename the Datacenter
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: present
+    data:
+        name: "MyDatacenter"
+        newName: "My Datacenter"
+  delegate_to: localhost
+
+- name: Remove the Datacenter
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: absent
+    data:
+        name: 'My Datacenter'
+  delegate_to: localhost
+'''
+
+RETURN = '''
+datacenter:
+    description: Has the OneView facts about the Datacenter.
+    returned: On state 'present'. Can be null.
+    type: dict
+'''
+
+from ansible.module_utils.oneview import OneViewModuleBase, OneViewModuleResourceNotFound
+
+
+class DatacenterModule(OneViewModuleBase):
+    MSG_CREATED = 'Datacenter added successfully.'
+    MSG_UPDATED = 'Datacenter updated successfully.'
+    MSG_ALREADY_PRESENT = 'Datacenter is already present.'
+    MSG_DELETED = 'Datacenter removed successfully.'
+    MSG_ALREADY_ABSENT = 'Datacenter is already absent.'
+    MSG_RACK_NOT_FOUND = 'Rack was not found.'
+    RESOURCE_FACT_NAME = 'datacenter'
+
+    argument_spec = dict(
+        state=dict(type='str', default='present', choices=['absent', 'present']),
+        data=dict(type='dict', required=True)
+    )
+
+    def __init__(self):
+        super(DatacenterModule, self).__init__(additional_arg_spec=self.argument_spec,
+                                               validate_etag_support=True)
+        self.resource_client = self.oneview_client.datacenters
+
+    def execute_module(self):
+
+        resource = self.get_by_name(self.data['name'])
+
+        if self.state == 'present':
+            self.__replace_name_by_uris(self.data)
+            return self.resource_present(resource, self.RESOURCE_FACT_NAME, 'add')
+        elif self.state == 'absent':
+            return self.resource_absent(resource, 'remove')
+
+    def __replace_name_by_uris(self, resource):
```

Comment:
any particular reason for the double underscores?

Reply:
No, it was the standard used when the module was first created but we are changing it to single underscore when delivering to ansible core. 
This indeed should be changed to single underscore format.

Reply:
it doesn't really matter as modules are treated as 'independent scripts' and not meant to be shared or reused code